src/sdt.rs

```diff
@@ -130,6 +131,17 @@ pub(crate) fn dispatch_sdt<H>(handler : &mut H, physical_address : usize) -> Res
          */
         match signature
         {
+            "FADT" => {
+                let fadt_mapping = handler.map_physical_region::<Fadt>(physical_address);
+                match parse_fadt(fadt_mapping) {
```

Comment:
I think it'd be better to lend the mapping to `parse_fadt` and then unmap it here, so the mapping and unmapping logic is in one place

Reply:
Not sure I understood what you meant

Reply:
So take a `&PhysicalMapping<Fadt>` instead in `parse_fadt`, change this match to `parse_fadt(&fadt_mapping)` and then do `handler.unmap_physical_region(fadt_mapping)` after the match